docs/development/pigs/pig-029.rst

```diff
@@ -0,0 +1,207 @@
+
+Currently, the `MapDataset` and the `FluxPointsDataset` support additional axes, so the main changes are required in the `MapEvaluator` and in the model definitions
+
+
+Use cases
+=========
+
+1. Fitting other parameters like phase, frequency, etc
+2. Reduction of code duplication: Analytic shapes like the power law is defined separately for `PowerLawSpectralModel`, `PowerLawTeLmporalModel` and `PowerLawNormSpectralModel`, which differ only in the unit handling
+3. Separation of units and values: This is a side benefit which can be useful if move to using JAX arrays in the future.
+
+
+Proposed API
+============
+
+We propose to have a general unit-less analytic class definition, inheriting from `ModelBase` The unit handling responsibility lies on the user.
+Internal support and checks and be provided for common input parameters (eg: time/energy) and the expected output units. 
+
+So, we move from 
+
+`SkyModel(spectral_model=, temporal_model=)`
+to 
+
+`SkyModelGeneral(submodels={"energy_true": p, "time":t})`
+
+where the user defines the axes the models are supposed affect.
+
+
+For a simple power law, it will be:
+
+"$\\phi(x) = \\phi_0 \\cdot \\left( \\frac{x-x_{ref}}{x_0} \\right)^{-\\alpha}$\n",
+
+
+```
+def integrate_linear(func, range_min, range_max, ndecade=100, oversampling_factor=100, **kwargs):
+        if isinstance(range_min, Time):
+            range_min=range_min.mjd * u.d
+            range_max=range_max.mjd * u.d
+        x, steps = np.linspace(
+            range_min, range_max, oversampling_factor, retstep=True, axis=-1
+        )
+        values = func(x)
+        integral = np.sum(values, axis=-1) * steps
+        return integral/np.sum(range_max - range_min) 
+
+class PowerLawModel(ModelBase):
+
+    tag = "Generalised_po"
+    alpha = Parameter("alpha", 2, frozen=True)
+    x0 = Parameter("x0", 1, frozen=True)
+    x_ref = Parameter("x_ref", 1, frozen=True)
+    phi0 = Parameter("phi0", 1, frozen=True, scale_method="scale10",
+                    interp="log")
+    
+    def __init__(self, input_units="", norm_units="", **kwargs):
+        super().__init__()
+        par_input_units = ["x0", "x_ref"]
+        self.axis_units = input_units
+        for p in par_input_units:
+            par = getattr(self, p)
+            par.unit = input_units
+        self.phi0.unit = norm_units
+
+    def __call__(self, x):
+        kwargs = {par.name: par.quantity for par in self.parameters}
+        if isinstance(x, Time):
+            x=Time(x, scale=x.scale).mjd * u.d
+        x = x.to(self.axis_units)
+        return self.evaluate(x, **kwargs)
+    
+    @staticmethod
+    def evaluate(x, x_ref, x0, alpha, phi0):
+        """Evaluate the model (static function)"""
+        return phi0*np.power((x-x_ref)/x0, -alpha)
```

Comment:
did we discuss at some point to have evaluate the unit separately ?

Reply:
Indeed we did!
So should we work like `np.power((x.value-x_ref.value)/x0.value, -alpha)` or should the parameters be devoid of units and only attached in the end?
The latter would required modifications in the `Parameter` class